frappe/core/doctype/doctype/doctype.py

```diff
@@ -3,1004 +3,1003 @@
-		self.validate_document_type()
-		validate_fields(self)
+class UniqueFieldnameError(frappe.ValidationError): pass
 
-		if self.istable:
-			# no permission records for child table
-			self.permissions = []
-		else:
-			validate_permissions(self)
 
-		self.make_amendable()
-		self.make_repeatable()
-		self.validate_nestedset()
-		self.validate_website()
+class IllegalMandatoryError(frappe.ValidationError): pass
 
-		if not self.is_new():
-			self.before_update = frappe.get_doc('DocType', self.name)
 
-		if not self.is_new():
-			self.setup_fields_to_fetch()
+class DoctypeLinkError(frappe.ValidationError): pass
 
-		if self.default_print_format and not self.custom:
-			frappe.throw(_('Standard DocType cannot have default print format, use Customize Form'))
 
+class WrongOptionsDoctypeLinkError(frappe.ValidationError): pass
 
-	def set_default_in_list_view(self):
-		'''Set default in-list-view for first 4 mandatory fields'''
-		if not [d.fieldname for d in self.fields if d.in_list_view]:
-			cnt = 0
-			for d in self.fields:
-				if d.reqd and not d.hidden and not d.fieldtype in table_fields:
-					d.in_list_view = 1
-					cnt += 1
-					if cnt == 4: break
 
+class HiddenAndMandatoryWithoutDefaultError(frappe.ValidationError): pass
 
-	def set_default_translatable(self):
-		'''Ensure that non-translatable never will be translatable'''
-		for d in self.fields:
-			if d.translatable and not supports_translation(d.fieldtype):
-				d.translatable = 0
 
+class NonUniqueError(frappe.ValidationError): pass
 
-	def check_developer_mode(self):
-		"""Throw exception if not developer mode or via patch"""
-		if frappe.flags.in_patch or frappe.flags.in_test:
-			return
 
-		if not frappe.conf.get("developer_mode") and not self.custom:
-			frappe.throw(_("Not in Developer Mode! Set in site_config.json or make 'Custom' DocType."), CannotCreateStandardDoctypeError)
+class CannotIndexedError(frappe.ValidationError): pass
 
 
-	def setup_fields_to_fetch(self):
-		'''Setup query to update values for newly set fetch values'''
-		try:
-			old_meta = frappe.get_meta(frappe.get_doc('DocType', self.name), cached=False)
-			old_fields_to_fetch = [df.fieldname for df in old_meta.get_fields_to_fetch()]
-		except frappe.DoesNotExistError:
-			old_fields_to_fetch = []
+class CannotCreateStandardDoctypeError(frappe.ValidationError): pass
 
-		new_meta = frappe.get_meta(self, cached=False)
 
-		self.flags.update_fields_to_fetch_queries = []
+form_grid_templates = {
+    "fields": "templates/form_grid/fields.html"
+}
 
-		if set(old_fields_to_fetch) != set([df.fieldname for df in new_meta.get_fields_to_fetch()]):
-			for df in new_meta.get_fields_to_fetch():
-				if df.fieldname not in old_fields_to_fetch:
-					link_fieldname, source_fieldname = df.fetch_from.split('.', 1)
-					link_df = new_meta.get_field(link_fieldname)
 
-					if frappe.db.db_type == 'postgres':
-						update_query = '''
+class DocType(Document):
+    def get_feed(self):
+        return self.name
```

Comment:
@IslamMesha We use tabs for indentation.
https://github.com/frappe/erpnext/wiki/Coding-Standards#use-simple-indenting

Also, it is better to raise a small PR with a focused fix. 
It also makes it easier to review.